src/fides/core/pull.py

```diff
@@ -66,6 +68,40 @@ def pull_existing_resources(
     return existing_keys
 
 
+def pull_resource_by_key(
+    manifests_dir: str,
+    url: str,
+    headers: Dict[str, str],
+    fides_key: str,
+    resource_type: str,
+) -> None:
+    """
+    Pull a resource from the server by its fides_key and update the local manifest file.
+    """
+    if manifests_dir[-1] == "/":
+        manifests_dir = manifests_dir[:-1]
+    manifest_path = f"{manifests_dir}/{resource_type}.yaml"
+    print(f"Pulling resource with fides_key: {fides_key}...", end=" ")
+    server_resource = get_server_resource(url, resource_type, fides_key, headers)
+    print("done.")
+
+    if server_resource:
+        try:
+            manifest = load_yaml_into_dict(manifest_path)
+        except FileNotFoundError:
+            print(
+                f"Manifest file {manifest_path} does not already exist and will be created"
+            )
+            manifest = {}
+        print("Writing out resource to file...", end=" ")
+        manifest[resource_type] = [server_resource]
+        write_manifest_file(manifest_path, manifest)
+        print("done.")
+
+    else:
+        echo_red(f"Resource with fides_key: {fides_key} not found on the server.")
```

Comment:
This error message can be confusing if "resource_type" didn't match what was expected with the fides key

Reply:
I can see that, I changed here to always say the resource type, does that seem like it will cut it? https://github.com/ethyca/fides/pull/5260/commits/828cbe872ae6a3ff6ade8bb366a9175bc1df8b39

Reply:
helpful thanks